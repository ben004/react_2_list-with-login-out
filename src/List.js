import React, { Component } from 'react'
import _ from 'lodash';
import history from './history'
let array=[];
export default class List extends Component {
  constructor(props) {
    super(props);
    this.state = {
      inputValue: '',
      items: []
    }
  }

  onInputChange(e) {
    this.setState({
      inputValue: e.target.value,
    });
  }

  addItem() {
    let items = this.state.items;
    items.push(this.state.inputValue);
    
    this.setState({
      items
    });
  }
  delItem()
  { 
    let items = this.state.items
    let update=[]
    update=_.pullAt(items,array);
    update.pop()
    array=[];
    this.setState({
      items
    });
  }
  handlechange(e,id)
  {
    array.push(id)
  }

  listItems() {
    let items = this.state.items;
    return (
      <ol>
        {
            
          items.map((val, id) => {
            return (
              <li key={id}>
                <input  type="checkbox" name="isDone"ref="check_me" checked={items.isDone} onChange={(e)=>this.handlechange(e,id)} />
                  {val}
              </li>
            );
          })
        }
      </ol>
      
    );
  }
handleLogout(){
    history.push('/')
}


    render() {
        return (
            <div>
              
                <h1>This is your Dashboard</h1>
                <input type="text" onChange={ (e) => this.onInputChange(e) } />
                <button data-testid="add_button" onClick={ () => this.addItem() }>Add item</button>
                <button data-testid="delete_button" onClick={ () => this.delItem() }>delete item</button>
                <br/>
                { this.listItems() }
                <br/>
                <button data-testid="logout_button" onClick={(e=>this.handleLogout(e))}>logout</button>
                
            </div>
        )
    }
}

