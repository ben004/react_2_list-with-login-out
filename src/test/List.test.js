import React from 'react'
import {render, cleanup} from '@testing-library/react'
import List from '../List'
const renderComponent = (props = {}) => 
  render(<List {...props}>list</List>)
 afterEach(cleanup)
 describe('list ', () =>{
      it('should match a snapshot', () => {
      const { asFragment } = renderComponent()
      
      expect(asFragment(<List />)).toMatchSnapshot()
     });
     it('should be enabled', () => {
        const { getByTestId } = renderComponent();
        expect(getByTestId("add_button").textContent).toBe("Add item");
        expect(getByTestId("delete_button").textContent).toBe("delete item");
        expect(getByTestId("logout_button").textContent).toBe("logout");
      });
});
   
