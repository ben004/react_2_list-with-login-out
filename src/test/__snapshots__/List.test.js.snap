// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list  should match a snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      This is your Dashboard
    </h1>
    <input
      type="text"
    />
    <button
      data-testid="add_button"
    >
      Add item
    </button>
    <button
      data-testid="delete_button"
    >
      delete item
    </button>
    <br />
    <ol />
    <br />
    <button
      data-testid="logout_button"
    >
      logout
    </button>
  </div>
</DocumentFragment>
`;
